typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
             TypeScript rules outside of the recommended configuration
         */
-        // '@typescript-eslint/naming-convention': 'error',
-        // '@typescript-eslint/no-base-to-string': 'error',
-        // '@typescript-eslint/no-confusing-void-expression': 'error',
-        // '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
-        // '@typescript-eslint/no-unnecessary-condition': 'error',
-        // '@typescript-eslint/no-unnecessary-qualifier': 'error',
-        // '@typescript-eslint/no-unnecessary-type-arguments': 'error',
-        // '@typescript-eslint/non-nullable-type-assertion-style': 'error',
-        // '@typescript-eslint/prefer-includes': 'error',
-        // '@typescript-eslint/prefer-nullish-coalescing': 'error',
-        // '@typescript-eslint/prefer-readonly': 'error',
-        // '@typescript-eslint/prefer-readonly-parameter-types': 'error',
-        // '@typescript-eslint/prefer-reduce-type-parameter': 'error',
-        // '@typescript-eslint/prefer-string-starts-ends-with': 'error',
-        // '@typescript-eslint/promise-function-async': 'error',
-        // '@typescript-eslint/require-array-sort-compare': 'error',
-        // '@typescript-eslint/strict-boolean-expressions': 'error',
-        // '@typescript-eslint/switch-exhaustiveness-check': 'error'
+        '@typescript-eslint/naming-convention': 'error',
+        '@typescript-eslint/no-base-to-string': 'error',
+        '@typescript-eslint/no-confusing-void-expression': 'error',
+        '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
+        '@typescript-eslint/no-unnecessary-condition': 'error',
```

Comment:
We started talking about this in our meeting today but didn't get to the bottom of which compiler options are set for the examples. This could use more investigation.

Reply:
Following up, strict null checks both [remove null and undefined from the domain](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#--strictnullchecks) of every type and [require a check](https://www.typescriptlang.org/tsconfig#strictNullChecks) or non-null assertion to provide a gaurantee.

Reply:
We agreed in person that this rule should be disabled by default because most of our codebases don't enable strict null checks. If the rule is turned on in those codebases it would always throw an error saying the rule is not supported. @rajsite this is ready to close.